### ISSUE TITLE
Bug: Unexpected side effect in computed property but it's not a side effect

### ISSUE THREAD

**Issue by Kelin2025** [vuejs/eslint-plugin-vue#492]
**Tell us about your environment**

* **ESLint Version:** v4.19.1
* **eslint-plugin-vue Version:** v4.0.0
* **Node Version:** v9.9.0

**Please show your full configuration:**

```js
module.exports = {
  root: true,
  env: {
    browser: true,
    node: true
  },
  parserOptions: {
    parser: 'babel-eslint'
  },
  extends: [
    // https://github.com/vuejs/eslint-plugin-vue#priority-a-essential-error-prevention
    // consider switching to `plugin:vue/strongly-recommended` or `plugin:vue/recommended` for stricter rules.
    'plugin:vue/essential'
  ],
  // required to lint *.vue files
  plugins: [
    'vue'
  ],
  // add your custom rules here
  rules: {}
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
sortedCoins() {
  return [...this.coins].sort((a, b) => a - b)
}
```

**What did you expect to happen?**

No warnings here because I sort new array

**What actually happened? Please include the actual, raw output from ESLint.**
```
ERROR in ./components/widget/coin/WidgetCoinTopMoversList.vue

/Users/kelin/Work/coin-nuxt/components/widget/coin/WidgetCoinTopMoversList.vue
  87:14  error  Unexpected side effect in "sortedCoins" computed property  vue/no-side-effects-in-computed-properties

✖ 1 problem (1 error, 0 warnings)

 @ ./node_modules/babel-loader/lib?{"babelrc":false,"cacheDirectory":true,"presets":["vue-app"],"plugins":["ramda",["nanoutils",{"cjs":true}]]}!./node_modules/vue-loader/lib/selector.js?type=script&index=0!./pages/index.vue 63:0-87
 @ ./pages/index.vue
 @ ./.nuxt/router.js
 @ ./.nuxt/index.js
 @ ./.nuxt/client.js
 @ multi webpack-hot-middleware/client?name=client&reload=true&timeout=30000&path=/__webpack_hmr ./.nuxt/client.js
```

**Comment by michalsnik** [comment]
Hey @Kelin2025 I think I fixed this problem in one of the latest releases. Please check out `v4.7.0` and if it's still broken feel free to reopen this issue. I checked it however and it seems to be working fine now :)
